includes/REST_API/Stories_Controller.php

```diff
@@ -86,6 +86,28 @@ public function prepare_item_for_response( $post, $request ) {
 			$data['style_presets'] = is_array( $style_presets ) ? $style_presets : self::EMPTY_STYLE_PRESETS;
 		}
 
+		if ( in_array( 'preview_link', $fields, true ) ) {
+			// Based on https://github.com/WordPress/wordpress-develop/blob/8153c8ba020c4aec0b9d94243cd39c689a0730f7/src/wp-admin/includes/post.php#L1445-L1457
+			if ( 'draft' === $post->post_status || empty( $post->post_name ) ) {
+				$view_link = get_preview_post_link( $post );
+			} else {
+				if ( 'publish' === $post->post_status ) {
+					$view_link = get_permalink( $post );
+				} else {
+					if ( ! function_exists( 'get_sample_permalink' ) ) {
+						require_once ABSPATH . 'wp-admin/includes/post.php';
+					}
+
+					list ( $permalink ) = get_sample_permalink( $post->ID, $post->post_title, '' );
+
+					// Allow non-published (private, future) to be viewed at a pretty permalink, in case $post->post_name is set.
+					$view_link = str_replace( [ '%pagename%', '%postname%' ], $post->post_name, $permalink );
```

Comment:
Not sure I follow here. Is this needed? Which post statuses does apply to?

Originally I thought it could be as simple as this:

https://github.com/WordPress/wordpress-develop/blob/8153c8ba020c4aec0b9d94243cd39c689a0730f7/src/wp-includes/rest-api/endpoints/class-wp-rest-autosaves-controller.php#L410-L421